### ISSUE TITLE
GPflow.sgpr memory issue in very large dataset

### ISSUE THREAD

**Issue by DolanDack** [GPflow/GPflow#489]
Apologies if this is not an issue per se, but I am trying to get familiar with GPflow (as an alternative to GPML) and I stumble upon a memory error. 
Specifically, I have a very large dataset with 820,000 observations and 9 dimensions.
 For my model I am using a composite kernel (SEard + Noise):
```
k1 = GPflow.kernels.RBF(input_dim=9, ARD=True)
k2 = GPflow.kernels.White(input_dim=9)
k3 = k1 + k2
```
For the GP regression model I use the Sparse Gaussian Process Regression (sgpr), with 50 inducing points:
```
M = 50 # Inducing points
Z = x[:M, :].copy() # Inducing points
m = GPflow.sgpr.SGPR(x, y, k3, Z)
m.likelihood.variance = 0.01
```
When I try to optimise the parameters `m.optimize()` I get his error:

> ResourceExhaustedError (see above for traceback): OOM when allocating tensor with shape[50,819621]

> 	 [[Node: gradients/model.build_likelihood/MatMul_grad/MatMul_1 = MatMul[T=DT_DOUBLE, transpose_a=true, transpose_b=false, _device="/job:localhost/replica:0/task:0/gpu:0"](gradients/AddN_1/_75, model.build_likelihood/truediv)]]
> 	 [[Node: grad_objective/_91 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_1369_grad_objective", tensor_type=DT_DOUBLE, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

I even tried to include mini-batches but I still get the same message
```
m.X.minibatch_size = 50
m.Y.minibatch_size = 50

m.Z.fixed = True
m.optimize()
```

Again, I am sorry for posting this question here. But, I would be grateful if someone has any suggestions on the matter. 
My pc specs are: 

- GPU 780ti with 11gb GPU-RAM; 
- tensorflow version = tensorflow-gpu 1.2.1; 
- Windows 10

**Comment by markvdw** [comment]
This should be possible. I haven't checked this, but try setting `minibatch_size` in the `SGPR` constructor. It's what I always do, and I've never had a problem like this.

**Comment by mrksr** [comment]
If I'm not mistaken, `SGPR`, which implements the "original" Titsias bound, cannot handle minibatches. With a huge training set, [kernel matrices involving `X`](https://github.com/GPflow/GPflow/blob/d3fd05eb99b3efacc4a4b558b7f33df63fa26330/GPflow/sgpr.py#L77) will require a rather large amount of memory and therefore fill the 11GB rather quickly.

If you want to use a model which can handle minibatches, [`SVGP`](https://github.com/GPflow/GPflow/blob/master/GPflow/svgp.py) is probably the correct choice. There, you can set the `minibatch_size` via a constructor parameter as mentioned by @markvdw.

**Comment by DolanDack** [comment]
Thank you very much guys!
